### ISSUE TITLE
Invalid URI error when opening AL workspace

### ISSUE THREAD

**Issue by TheDenSter** [Microsoft/AL#3983]
Copied the messages below from the output window in VSCode when opening any AL workspace, looks like a bunch of 'invalid URI' messages. Pulled the latest bcsandbox image this morning, refreshed my container, everything was working. Then I got the message to update VSCode, and after that it all went to crap. Noticed that I was on an old version of the AL Language extension, so updated that as well (2.0.48254). 

[Error - 11:46:43 AM] Please report this issue to https://github.com/microsoft/al/issues including information on how to reproduce it, if possible.
Processing of message 'workspace/didChangeConfiguration' failed with error: 'Invalid URI: The format of the URI could not be determined.'
Details:
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Utilities.PathUtilities.CreateUriFilePath(String filePath) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Utilities\PathUtilities.cs:line 417
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.CreateProjectInfo(String projectPath, String assemblyName, CompilationOptions options, ParseOptions parseOptions, String packageCachePath, String[] probingFolders, String ruleSetPath) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 630
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.SetCurrentProjectFromPath(String projectPath, String packageCachePath, String[] assemblyProbingPaths, String ruleSetPath, List`1& projectLoadingDiagnostics) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 139
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.TrySetCurrentProject(String projectPath, String packageCachePath, String[] assemblyProbingPaths, String ruleSetPath, IList`1& diagnostics, Boolean& missingManifest, Boolean& missingSymbols) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 71
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.ProjectRequestHandler`1.TrySetCurrentProject(SettingsRequest request, CancellationToken cancellationToken) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\ProjectRequestHandler.cs:line 31
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.DidChangeConfigurationRequestHandler.HandleAsync(SettingsRequest request, Int32 requestId, CancellationToken cancellationToken) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\DidChangeConfigurationRequestHandler.cs:line 49
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.MessageProtocol.RequestHandlerBase`1.<HandleAsync>d__10.MoveNext() in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\MessageProtocol\RequestHandlerBase.cs:line 45
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.RequestRegistry.<Process>d__6.MoveNext() in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\Endpoints\RequestRegistry.cs:line 59

[Error - 11:46:43 AM] Please report this issue to https://github.com/microsoft/al/issues including information on how to reproduce it, if possible.
Processing of message 'al/setActiveWorkspace' failed with error: 'Invalid URI: The format of the URI could not be determined.'
Details:
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Utilities.PathUtilities.CreateUriFilePath(String filePath) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Utilities\PathUtilities.cs:line 417
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.CreateProjectInfo(String projectPath, String assemblyName, CompilationOptions options, ParseOptions parseOptions, String packageCachePath, String[] probingFolders, String ruleSetPath) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 630
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.SetCurrentProjectFromPath(String projectPath, String packageCachePath, String[] assemblyProbingPaths, String ruleSetPath, List`1& projectLoadingDiagnostics) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 139
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.TrySetCurrentProject(String projectPath, String packageCachePath, String[] assemblyProbingPaths, String ruleSetPath, IList`1& diagnostics, Boolean& missingManifest, Boolean& missingSymbols) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 71
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.ProjectRequestHandler`1.TrySetCurrentProject(SettingsRequest request, CancellationToken cancellationToken) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\ProjectRequestHandler.cs:line 31
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.Extensions.SetActiveWorkspaceRequestHandler.HandleAsync(SettingsRequest request, Int32 requestId, CancellationToken cancellationToken) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\Extensions\SetActiveWorkspaceRequestHandler.cs:line 24
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.MessageProtocol.RequestHandlerBase`1.<HandleAsync>d__10.MoveNext() in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\MessageProtocol\RequestHandlerBase.cs:line 45
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.RequestRegistry.<Process>d__6.MoveNext() in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\Endpoints\RequestRegistry.cs:line 59

[Error - 11:46:44 AM] Please report this issue to https://github.com/microsoft/al/issues including information on how to reproduce it, if possible.
Processing of message 'al/setActiveWorkspace' failed with error: 'Invalid URI: The format of the URI could not be determined.'
Details:
System.UriFormatException: Invalid URI: The format of the URI could not be determined.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Utilities.PathUtilities.CreateUriFilePath(String filePath) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Utilities\PathUtilities.cs:line 417
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.CreateProjectInfo(String projectPath, String assemblyName, CompilationOptions options, ParseOptions parseOptions, String packageCachePath, String[] probingFolders, String ruleSetPath) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 630
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.SetCurrentProjectFromPath(String projectPath, String packageCachePath, String[] assemblyProbingPaths, String ruleSetPath, List`1& projectLoadingDiagnostics) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 139
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.VsCodeWorkspace.TrySetCurrentProject(String projectPath, String packageCachePath, String[] assemblyProbingPaths, String ruleSetPath, IList`1& diagnostics, Boolean& missingManifest, Boolean& missingSymbols) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\VsCodeWorkspace.cs:line 71
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.ProjectRequestHandler`1.TrySetCurrentProject(SettingsRequest request, CancellationToken cancellationToken) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\ProjectRequestHandler.cs:line 31
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.Extensions.SetActiveWorkspaceRequestHandler.HandleAsync(SettingsRequest request, Int32 requestId, CancellationToken cancellationToken) in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\Extensions\SetActiveWorkspaceRequestHandler.cs:line 24
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.MessageProtocol.RequestHandlerBase`1.<HandleAsync>d__10.MoveNext() in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\MessageProtocol\RequestHandlerBase.cs:line 45
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.RequestRegistry.<Process>d__6.MoveNext() in C:\a\459\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\Endpoints\RequestRegistry.cs:line 59

**Comment by TheDenSter** [comment]
I just created a new AL:Go workspace, and that works fine. Comparing the two workspaces, here are the differences:
- New workspace has breakOnError turned on in launch.json
- Platform/application both set to 13 instead of 12
- New workspace has added 'runtime'
Everything else is identical.
Then, I changed the old workspace to have the same values as the new workspace, and I am still getting the 'invalid URI' error, even when re-creating the container in between tries.

**Comment by TheDenSter** [comment]
Welp, throw me out because I can't even reproduce it anymore. I had 3 workspaces. Two were created this morning with the older version of VSCode, both had (and still do have) the problem, even when I update the launch.json and app.json. The 3rd workspace I created with the new version of the AL:Go workspace that I just created. So then I thought I would have to re-create all of my demo and workshop workspaces, so as a test I tried opening a few of those demo workspaces, and each one that I've tried so far works without a problem.
No clue what is going on, but it doesn't seem to be a problem now. I've deleted the two workspaces that don't work, so no problem anymore. I'll close this issue
